crates/rsonpath-lib/src/classification/structural/avx2.rs

```diff
@@ -63,9 +63,11 @@ impl Iterator for StructuralsBlock<'_> {
 
             match self.quote_classified.block[idx] {
                 b':' => Colon(idx),
-                b'[' | b'{' => Opening(idx),
+                b'{' => OpeningBrace(idx),
+                b'[' => OpeningBracket(idx),
                 b',' => Comma(idx),
-                _ => Closing(idx),
+                b'}' => ClosingBrace(idx),
+                _ => ClosingBracket(idx),
```

Comment:
The implementation used in the analogous portion of `nosimd` seems preferable, as it would identify missing cases when the enum is added to in the future.  is there a reason to prefer this over something like:
```suggestion
                b']' => ClosingBracket(idx),
                c => unreachable!("'{c}' was specified as a structural character but is not implemented.")
```

Reply:
It's measurably slower, actually. The jump table is compiled differently, and this is a hot path so the impact is big.

![image](https://user-images.githubusercontent.com/20441980/232617801-80d9cbc1-a070-4da7-92d0-959d31d4054f.png)